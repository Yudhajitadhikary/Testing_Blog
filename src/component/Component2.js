import React, { Component } from 'react';
import Component1 from './Component1'
class Component2 extends Component {
    constructor(props) {
        super(props);
       
        this.Click = this.Click.bind(this);
        }
        Click() {
           this.props.action()
          }
    render() {
        const {count}=this.props
        return (
            <div>
            <button id='button' onClick={this.Click}>Count Total</button>
               <div>{count}</div>
               <Component1/>
            </div>
        );
    }
}

export default Component2;