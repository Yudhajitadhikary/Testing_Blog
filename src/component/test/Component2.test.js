import React from "react";
import { mount } from "../enzyme";
import { shallowToJson } from "enzyme-to-json";
import Component2 from '../Component2';
import { Provider } from 'react-redux';
import store from '../../store';
 let props={
     action:jest.fn()
 }
let wrapper;
it("should render TextFieldAtom component", () => {
  wrapper = mount(<Provider store={store}><Component2 {...props}/></Provider>);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
//   console.log(shallowToJson(wrapper).children[0].children[0].children[0])
  wrapper.find('#button').simulate('click')

});

