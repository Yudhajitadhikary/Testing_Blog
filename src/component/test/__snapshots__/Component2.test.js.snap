// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TextFieldAtom component 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Component2
    action={[MockFunction]}
  >
    <div>
      <button
        id="button"
        onClick={[Function]}
      >
        Count Total
      </button>
      <div />
      <Component1>
        <div
          style={
            Object {
              "display": "flex",
            }
          }
        >
          <div>
            <div>
              <div>
                Yudhajit Adhikary
              </div>
              <div>
                Koyena Adhikary
              </div>
            </div>
          </div>
          <div>
            <div>
              <div>
                Blog
              </div>
              <div>
                Blog1
              </div>
            </div>
          </div>
        </div>
      </Component1>
    </div>
  </Component2>
</Provider>
`;
